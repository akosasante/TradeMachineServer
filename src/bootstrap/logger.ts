import fs from "fs";
import path from "path";
import winston from "winston";

const { combine, timestamp, printf, json, colorize, align } = winston.format;
// No need to write error if we're only printing that level here anyway
const errorFileLogFormat = printf(info => {
    return `[ ${info.timestamp} ]:\t${info.message}`;
});
const timestampFormat = "YYYY-MM-DD hh:mm:ss A";

const alignedWithColorsAndTime = combine(
  colorize(),
  timestamp(),
  align(),
  printf(info => `[ ${info.timestamp}] ${info.level}: ${info.message}`)
);

// Transport objects
const consoleLogger = new winston.transports.Console({
    format: alignedWithColorsAndTime,
});

let errorLogger;
let combinedLogger;
let exceptionHandler;

if (process.env.NODE_ENV !== "test") {
    const logDir = path.resolve(`${process.env.BASE_DIR}/logs`);

    if (!fs.existsSync(logDir)) {
        fs.mkdirSync(logDir);
    }

    errorLogger = new winston.transports.File({
        filename: `${logDir}/server-error.log`,
        level: "error",
        eol: "\r\n",
        maxsize: 52428800,
        maxFiles: 10,
        tailable: true,
    });

    combinedLogger = new winston.transports.File({
        filename: `${logDir}/server-combined.log`,
        level: "info",
        eol: "\r\n",
        format: json(),
        maxsize: 52428800,
        maxFiles: 10,
        tailable: true,
    });

    exceptionHandler = new winston.transports.File({
        filename: `${logDir}/uncaught-exceptions.log`,
        eol: "\r\n",
        maxsize: 52428800,
        maxFiles: 10,
        tailable: true,
    });
}
// Logger object
const logger = winston.createLogger({
    level: "debug",
    format: combine(timestamp({ format: timestampFormat }), errorFileLogFormat),
    transports: errorLogger && combinedLogger ? [errorLogger, combinedLogger] : [],
    silent: process.env.ENABLE_LOGS === "false",
    exceptionHandlers: exceptionHandler ? [exceptionHandler] : [],
    exitOnError: false,
});

if (process.env.ENABLE_LOGS === "true") {
    logger.add(consoleLogger);
}

// Last ditch if the logger itself errors
// eslint-disable-next-line no-console
logger.on("error", err => console.error(err));

export default logger;
