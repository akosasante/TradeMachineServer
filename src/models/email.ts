import { Column, CreateDateColumn, Entity, Index, ManyToOne, PrimaryColumn, UpdateDateColumn } from "typeorm";
import Trade from "./trade";

@Entity()
export default class Email {
    @PrimaryColumn()
    public messageId!: string;

    @Column()
    @Index()
    public status?: string;

    @CreateDateColumn()
    public dateCreated?: Date;

    @UpdateDateColumn()
    public dateModified?: Date;

    @ManyToOne(_type => Trade, trade => trade.tradeParticipants, { onDelete: "SET NULL", nullable: true })
    public trade!: Trade;

    constructor(props: Partial<Email> & Required<Pick<Email, "messageId">>) {
        Object.assign(this, props);
    }

    public toString(): string {
        return `${this.constructor.name}#${this.messageId}`;
    }

    public parse<T extends Email>(): Partial<T> {
        return Object.assign({}, this);
    }
}
