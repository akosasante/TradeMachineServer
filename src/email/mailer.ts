import Email from "email-templates";
import nodemailer from "nodemailer";
// @ts-ignore
import SendinBlueTransport from "nodemailer-sendinblue-transport";
import path from "path";
import { inspect } from "util";
import logger from "../bootstrap/logger";
import User from "../models/user";
import { config as dotenvConfig } from "dotenv";
dotenvConfig({path: path.resolve(__dirname, "../../.env")});

export interface SendInBlueSendResponse {
    envelope: {
        from: string;
        to: string[];
    };
    messageId?: string;
    message: string;
    originalMessage: {
        to: string;
        from: string;
        subject: string;
        html: string;
        text: string;
        attachments: object[];
    };
}

const SendInBlueOpts = {
    apiKey: process.env.EMAIL_KEY,
    apiUrl: process.env.EMAIL_API_URL,
};

const SendInBlueTransport = nodemailer.createTransport(SendinBlueTransport(SendInBlueOpts));

const baseDomain = process.env.BASE_URL;

export const Emailer = {
    emailer: new Email({
        juice: true,
        juiceResources: {
            webResources: {
                relativeTo: path.resolve("./src/email/templates"),
                images: false,
            },
        },
        message: {
            from: "tradebot@flexfoxfantasy.com",
        },
        subjectPrefix: "FlexFoxFantasy TradeMachine - ",
        transport: SendInBlueTransport,
        views: {
            root: path.resolve("./src/email/templates"),
        },
    }),

    async sendPasswordResetEmail(user: User): Promise<SendInBlueSendResponse> {
        const resetPassPage = `${baseDomain}/reset_password?u=${encodeURI(user.id!)}`;
        logger.debug("sending password reset email");
        return Emailer.emailer.send({
            template: "reset_password",
            message: {
                to: user.email,
            },
            locals: {
                name: user.displayName || user.email,
                url: resetPassPage,
            },
        })
        .then((res: any) => {
            logger.info(`Successfully sent password reset email: ${inspect(res.messageId)}`);
            return res;
        })
        .catch((err: Error) => {
            logger.error(`Ran into an error while sending password reset email: ${inspect(err)}`);
            return undefined;
        });
    },

    async sendTestEmail(user: User): Promise<SendInBlueSendResponse> {
        logger.debug(`sending test email to user: ${user}`);
        return Emailer.emailer.send({
            template: "test_email",
            message: {
                to: user.email,
                subject: "Test Email",
            },
            locals: {
                name: user.displayName || user.email,
            },
        })
        .then((res: any) => {
            logger.info(`Successfully sent test email: ${inspect(res.messageId)}`);
            return res;
        })
        .catch((err: Error) => {
            logger.error(`Ran into an error while sending test email: ${inspect(err)}`);
            return undefined;
        });
    },

    async sendRegistrationEmail(user: User): Promise<SendInBlueSendResponse> {
        logger.debug("sending registration email");
        const registrationLink = `${baseDomain}/register`;
        return Emailer.emailer.send({
            template: "registration_email",
            message: {
                to: user.email,
            },
            locals: {
                name: user.displayName || user.email,
                url: registrationLink,
            },
        })
        .then((res: any) => {
            logger.info(`Successfully sent registration email: ${inspect(res.messageId)}`);
            return res;
        })
        .catch((err: Error) => {
            logger.error(`Ran into an error while sending registration email: ${inspect(err)}`);
            return undefined;
        });
    },
};

Object.freeze(Emailer);

async function test() {
    const mailer = Emailer;
    const user = new User({displayName: "Akosua", email: "tripleabatt@gmail.com"});
    logger.info("BEFORE");
    const res = await mailer.sendTestEmail(user);
    logger.info(`RESULT: ${inspect(res)}`);
}
test();
