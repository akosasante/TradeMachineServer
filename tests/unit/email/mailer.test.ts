import "jest";
import "jest-extended";
import {Emailer} from "../../../src/email/mailer";
import {UserFactory} from "../../factories/UserFactory";
import logger from "../../../src/bootstrap/logger";
import {TradeFactory} from "../../factories/TradeFactory";
import {DraftPickFactory} from "../../factories/DraftPickFactory";
import {TeamFactory} from "../../factories/TeamFactory";
import {PlayerFactory} from "../../factories/PlayerFactory";
import {PlayerLeagueType} from "../../../src/models/player";

describe("Emailer Class", () => {
    beforeAll(() => {
        logger.debug("~~~~~~EMAILER TESTS BEGIN~~~~~~");
    });
    afterAll(() => {
        logger.debug("~~~~~~EMAILER TESTS COMPLETE~~~~~~");
    });
    const testUser = UserFactory.getUser("test@example.com", "Jatheesh", undefined, undefined, {id: "test-uuid", passwordResetToken: "random-token"});
    const testItems = [
        TradeFactory.getTradedMajorPlayer(PlayerFactory.getPlayer(undefined, PlayerLeagueType.MAJOR, {mlbTeam: "Pirates", meta: {espnPlayer: {player: {eligibleSlots: [0, 1, 7]}}}})),
        TradeFactory.getTradedMinorPlayer(PlayerFactory.getPlayer("MiniMe", undefined, {meta: {minorLeaguePlayerFromSheet: {mlbTeam: "Jays", position: "P", leagueLevel: "Low"}}})),
        TradeFactory.getTradedPick(DraftPickFactory.getPick(undefined, undefined, undefined, undefined, { originalOwner: TeamFactory.getTeam() })),
    ];
    const testTrade = TradeFactory.getTrade( testItems, undefined, undefined, {id: "test-uuid"});

    describe("email snapshots", () => {
        it("sendTestEmail", async () => {
            const res = await Emailer.sendTestEmail(testUser);
            delete res.message; // value varies due to include a dynamic messageId so keep it out of the snapshot
            delete res.messageId;
            expect(res).toMatchSnapshot();
        });

        it("sendRegistrationEmail", async () => {
            const res = await Emailer.sendRegistrationEmail(testUser);
            delete res.message; // value varies due to include a dynamic messageId so keep it out of the snapshot
            delete res.messageId;
            expect(res).toMatchSnapshot();
        });

        it("sendPasswordResetEmail", async () => {
            const res = await Emailer.sendPasswordResetEmail(testUser);
            delete res.message; // value varies due to include a dynamic messageId so keep it out of the snapshot
            delete res.messageId;
            expect(res).toMatchSnapshot();
        });

        it("sendTradeRequestEmail", async () => {
            const res = await Emailer.sendTradeRequestEmail("test@example.com", testTrade);
            delete res.message;
            delete res.messageId;
            expect(res).toMatchSnapshot();
        });

        it("sendTradeSubmissionEmail", async () => {
            const res = await Emailer.sendTradeSubmissionEmail("test@exaample.com", testTrade);
            delete res.message;
            delete res.messageId;
            expect(res).toMatchSnapshot();
        });
    });
});
