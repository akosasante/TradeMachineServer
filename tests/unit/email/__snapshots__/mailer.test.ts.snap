// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emailer Class email snapshots sendPasswordResetEmail 1`] = `
Object {
  "envelope": Object {
    "from": "tradebot@flexfoxfantasy.com",
    "to": Array [
      "test@example.com",
    ],
  },
  "originalMessage": Object {
    "attachments": Array [],
    "from": "tradebot@flexfoxfantasy.com",
    "html": "<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\" style=\\"height: 100%; background: #f7f7f7;\\">
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width; initial-scale=1.0;\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta name=\\"x-apple-disable-message-reformatting\\">
    
    <style>
@media screen and (max-width: 600px) {
  u + .body {
    width: 100vw !important;
  }
}
</style>
    <style>
a.button-link {
  background-color: #3b459a;
  display: inline-block;
  border: 2px solid transparent;
  color: #fafafa;
  padding: calc(.375em - 1px) .75em;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  border-radius: 4px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4);
  font-size: 1rem;
  text-decoration: none;
  white-space: nowrap;
  height: 1.5em;
  vertical-align: top;
  line-height: 1.5;
}
@media only screen and (max-width:700px) {
  .main,
.header,
.footer {
    width: 100% !important;
  }

  .main .container h1 {
    font-size: 1.8rem !important;
  }

  a.button-link {
    margin: 1rem 0 !important;
    text-decoration: underline !important;
  }
}
</style>
    <link href=\\"./reset_password/styles.css\\" rel=\\"stylesheet\\">
    <title>Trade Machine Password Reset</title>
  </head>
  <body style=\\"font-family: -apple-system, BlinkMacSystemFont, Roboto, Helvetica, Arial, sans-serif; font-size: 16px; color: black; height: 100%; background: #f7f7f7;\\">
    <header class=\\"title\\" style=\\"height: 100px; background-color: #a9c07c; background-image: url('https://imgur.com/VKEupAm.jpg'); background-size: cover; background-position: center center; background-repeat: no-repeat; width: 75%; margin: auto;\\"></header>
    <div class=\\"main\\" style=\\"height: 80%; background: #fff; width: 75%; margin-left: auto; margin-right: auto;\\">
      <div class=\\"container\\" style=\\"width: 70%; margin-left: 10%; padding: 5% 0;\\">
        <h1 style=\\"line-height: initial; margin: 0.67em 0; padding: 0; color: #3b459a; font-weight: 600; font-size: 3rem; font-family: 'Lucida Grande', 'Montserrat', 'Roboto', monospace;\\">TradeMachine</h1>
        <h2>Forgot Your Password?</h2>
        <p style=\\"line-height: 1.6em; margin: 1em 0;\\">Hi Jatheesh, you are receiving this email because you have requested a password reset for your account. <br><br></p>
        <p style=\\"line-height: 1.6em; margin: 1em 0;\\">Please click on the following link to complete the process:</p><a class=\\"button-link\\" href=\\"http://localhost:8080/reset_password?u=random-token\\" style=\\"background-color: #3b459a; display: inline-block; border: 2px solid transparent; color: #fafafa; padding: calc(.375em - 1px) .75em; text-align: center; white-space: nowrap; cursor: pointer; border-radius: 4px; box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4); font-size: 1rem; text-decoration: none; white-space: nowrap; height: 1.5em; vertical-align: top; line-height: 1.5;\\"><span>Reset Password</span></a>
        <div class=\\"divider\\"></div>
      </div>
    </div>
    <footer style=\\"height: 5%; background: #4b528b; width: 75%; margin-left: auto; margin-right: auto;\\"></footer>
  </body>
</html>",
    "subject": "FlexFoxFantasy TradeMachine - Password reset for Flex Fox Fantasy League TradeMachine",
    "text": " TRADEMACHINE
FORGOT YOUR PASSWORD?
Hi Jatheesh, you are receiving this email because you have requested a password
reset for your account. 



Please click on the following link to complete the process:

Reset Password [http://localhost:8080/reset_password?u=random-token]",
    "to": "test@example.com",
  },
}
`;

exports[`Emailer Class email snapshots sendRegistrationEmail 1`] = `
Object {
  "envelope": Object {
    "from": "tradebot@flexfoxfantasy.com",
    "to": Array [
      "test@example.com",
    ],
  },
  "originalMessage": Object {
    "attachments": Array [],
    "from": "tradebot@flexfoxfantasy.com",
    "html": "<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\" style=\\"height: 100%; background: #f7f7f7;\\">
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width; initial-scale=1.0;\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta name=\\"x-apple-disable-message-reformatting\\">
    
    <style>
@media screen and (max-width: 600px) {
  u + .body {
    width: 100vw !important;
  }
}
</style>
    <style>
a.button-link {
  background-color: #3b459a;
  display: inline-block;
  border: 2px solid transparent;
  color: #fafafa;
  padding: calc(.375em - 1px) .75em;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  border-radius: 4px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4);
  font-size: 1rem;
  text-decoration: none;
  white-space: nowrap;
  height: 1.5em;
  vertical-align: top;
  line-height: 1.5;
}
@media only screen and (max-width:700px) {
  .main,
.header,
.footer {
    width: 100% !important;
  }

  .main .container h1 {
    font-size: 1.8rem !important;
  }

  a.button-link {
    margin: 1rem 0 !important;
    text-decoration: underline !important;
  }
}
</style>
    <link href=\\"./registration_email/styles.css\\" rel=\\"stylesheet\\">
    <title>Trade Machine Registration Invite</title>
  </head>
  <body style=\\"font-family: -apple-system, BlinkMacSystemFont, Roboto, Helvetica, Arial, sans-serif; font-size: 16px; color: black; height: 100%; background: #f7f7f7;\\">
    <header class=\\"title\\" style=\\"height: 100px; background-color: #a9c07c; background-image: url('https://imgur.com/VKEupAm.jpg'); background-size: cover; background-position: center center; background-repeat: no-repeat; width: 75%; margin: auto;\\"></header>
    <div class=\\"main\\" style=\\"height: 80%; background: #fff; width: 75%; margin-left: auto; margin-right: auto;\\">
      <div class=\\"container\\" style=\\"width: 70%; margin-left: 10%; padding: 5% 0;\\">
        <h1 style=\\"line-height: initial; margin: 0.67em 0; padding: 0; color: #3b459a; font-weight: 600; font-size: 3rem; font-family: 'Lucida Grande', 'Montserrat', 'Roboto', monospace;\\">TradeMachine</h1>
        <h2>Invitation to Register</h2>
        <p style=\\"line-height: 1.6em; margin: 1em 0;\\">Hi Jatheesh, you have been invited to register on Trade Machine. This will allow you to submit trades with other members of the Flex Fox Fantasy Baseball League.</p>
        <p style=\\"line-height: 1.6em; margin: 1em 0;\\">Please click on the following link to complete the process:</p><a class=\\"button-link\\" href=\\"http://localhost:8080/register?e=dGVzdEBleGFtcGxlLmNvbQ==\\" style=\\"background-color: #3b459a; display: inline-block; border: 2px solid transparent; color: #fafafa; padding: calc(.375em - 1px) .75em; text-align: center; white-space: nowrap; cursor: pointer; border-radius: 4px; box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4); font-size: 1rem; text-decoration: none; white-space: nowrap; height: 1.5em; vertical-align: top; line-height: 1.5;\\"><span>Register</span></a>
        <div class=\\"divider\\"></div>
      </div>
    </div>
    <footer style=\\"height: 5%; background: #4b528b; width: 75%; margin-left: auto; margin-right: auto;\\"></footer>
  </body>
</html>",
    "subject": "FlexFoxFantasy TradeMachine - You have been invited to register on FFF Trade Machine",
    "text": " TRADEMACHINE
INVITATION TO REGISTER
Hi Jatheesh, you have been invited to register on Trade Machine. This will allow
you to submit trades with other members of the Flex Fox Fantasy Baseball League.

Please click on the following link to complete the process:

Register [http://localhost:8080/register?e=dGVzdEBleGFtcGxlLmNvbQ==]",
    "to": "test@example.com",
  },
}
`;

exports[`Emailer Class email snapshots sendTestEmail 1`] = `
Object {
  "envelope": Object {
    "from": "tradebot@flexfoxfantasy.com",
    "to": Array [
      "test@example.com",
    ],
  },
  "originalMessage": Object {
    "attachments": Array [],
    "from": "tradebot@flexfoxfantasy.com",
    "html": "<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\" style=\\"height: 100%; background: #f7f7f7;\\">
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width; initial-scale=1.0;\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta name=\\"x-apple-disable-message-reformatting\\">
    
    <style>
@media screen and (max-width: 600px) {
  u + .body {
    width: 100vw !important;
  }
}
</style>
    <style>
a.button-link {
  background-color: #3b459a;
  display: inline-block;
  border: 2px solid transparent;
  color: #fafafa;
  padding: calc(.375em - 1px) .75em;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  border-radius: 4px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4);
  font-size: 1rem;
  text-decoration: none;
  white-space: nowrap;
  height: 1.5em;
  vertical-align: top;
  line-height: 1.5;
}
@media only screen and (max-width:700px) {
  .main,
.header,
.footer {
    width: 100% !important;
  }

  .main .container h1 {
    font-size: 1.8rem !important;
  }

  a.button-link {
    margin: 1rem 0 !important;
    text-decoration: underline !important;
  }
}
</style>
    <title>Trade Machine Test Email</title>
  </head>
  <body style=\\"font-family: -apple-system, BlinkMacSystemFont, Roboto, Helvetica, Arial, sans-serif; font-size: 16px; color: black; height: 100%; background: #f7f7f7;\\">
    <header class=\\"title\\" style=\\"height: 100px; background-color: #a9c07c; background-image: url('https://imgur.com/VKEupAm.jpg'); background-size: cover; background-position: center center; background-repeat: no-repeat; width: 75%; margin: auto;\\"></header>
    <div class=\\"main\\" style=\\"height: 80%; background: #fff; width: 75%; margin-left: auto; margin-right: auto;\\">
      <div class=\\"container\\" style=\\"width: 70%; margin-left: 10%; padding: 5% 0;\\">
        <h1 style=\\"line-height: initial; margin: 0.67em 0; padding: 0; color: #3b459a; font-weight: 600; font-size: 3rem; font-family: 'Lucida Grande', 'Montserrat', 'Roboto', monospace;\\">TradeMachine</h1>
        <h2>Testing... Testing...</h2>
        <p style=\\"line-height: 1.6em; margin: 1em 0;\\">Hi Jatheesh, this is a test email.</p>
      </div>
    </div>
    <footer style=\\"height: 5%; background: #4b528b; width: 75%; margin-left: auto; margin-right: auto;\\"></footer>
  </body>
</html>",
    "subject": "FlexFoxFantasy TradeMachine - Test Email",
    "text": " TRADEMACHINE
TESTING... TESTING...
Hi Jatheesh, this is a test email.",
    "to": "test@example.com",
  },
}
`;

exports[`Emailer Class email snapshots sendTradeRequestEmail 1`] = `
Object {
  "envelope": Object {
    "from": "tradebot@flexfoxfantasy.com",
    "to": Array [
      "test@example.com",
    ],
  },
  "originalMessage": Object {
    "attachments": Array [],
    "from": "tradebot@flexfoxfantasy.com",
    "html": "<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\" style=\\"height: 100%; background: #f7f7f7;\\">
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width; initial-scale=1.0;\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta name=\\"x-apple-disable-message-reformatting\\">
    
    <style>
@media screen and (max-width: 600px) {
  u + .body {
    width: 100vw !important;
  }
}
</style>
    <style>
a.button-link {
  background-color: #3b459a;
  display: inline-block;
  border: 2px solid transparent;
  color: #fafafa;
  padding: calc(.375em - 1px) .75em;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  border-radius: 4px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4);
  font-size: 1rem;
  text-decoration: none;
  white-space: nowrap;
  height: 1.5em;
  vertical-align: top;
  line-height: 1.5;
}
@media only screen and (max-width:700px) {
  .main,
.header,
.footer {
    width: 100% !important;
  }

  .main .container h1 {
    font-size: 1.8rem !important;
  }

  a.button-link {
    margin: 1rem 0 !important;
    text-decoration: underline !important;
  }
}
</style>
    <style>
.button-container a.button-link {
  margin-right: 2rem !important;
}
</style>
    <title></title>
  </head>
  <body style=\\"font-family: -apple-system, BlinkMacSystemFont, Roboto, Helvetica, Arial, sans-serif; font-size: 16px; color: black; height: 100%; background: #f7f7f7;\\">
    <header class=\\"title\\" style=\\"height: 100px; background-color: #a9c07c; background-image: url('https://imgur.com/VKEupAm.jpg'); background-size: cover; background-position: center center; background-repeat: no-repeat; width: 75%; margin: auto;\\"></header>
    <div class=\\"main\\" style=\\"height: 80%; background: #fff; width: 75%; margin-left: auto; margin-right: auto;\\">
      <div class=\\"container\\" style=\\"width: 70%; margin-left: 10%; padding: 5% 0;\\">
        <h1 style=\\"line-height: initial; margin: 0.67em 0; padding: 0; color: #3b459a; font-weight: 600; font-size: 3rem; font-family: 'Lucida Grande', 'Montserrat', 'Roboto', monospace;\\">TradeMachine</h1>
        <h2>CREATOR_TEAM requested a trade with you:</h2>
        <div class=\\"divider\\"></div>
        <h3>CREATOR_TEAM would receive:</h3>
        <ul class=\\"picks\\" style=\\"margin-top: initial !important; margin-bottom: initial !important;\\"></ul>
        <li>Squirtle Squad's 2020 1st round Low Minors pick (#12) from RECIPIENT_TEAM</li>
        <hr>
        <h3>RECIPIENT_TEAM would receive:</h3>
        <ul class=\\"majors\\" style=\\"margin-top: initial !important; margin-bottom: initial !important;\\"></ul>
        <li>Honus Wiener (Pirates - C, 1B) from CREATOR_TEAM</li>
        <ul class=\\"minors\\" style=\\"margin-top: initial !important; margin-bottom: initial !important;\\"></ul>
        <li>MiniMe (Jays - P - Low Minors) from CREATOR_TEAM</li>
        <hr>
        <div class=\\"button-container\\"><a class=\\"button-link\\" href=\\"http://localhost:8080/trade/test-uuid/accept\\" style=\\"background-color: #3b459a; display: inline-block; border: 2px solid transparent; color: #fafafa; padding: calc(.375em - 1px) .75em; text-align: center; white-space: nowrap; cursor: pointer; border-radius: 4px; box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4); font-size: 1rem; text-decoration: none; white-space: nowrap; height: 1.5em; vertical-align: top; line-height: 1.5; margin-right: 2rem !important;\\"><span>Accept Trade</span></a><a class=\\"button-link\\" href=\\"http://localhost:8080/trade/test-uuid/reject\\" style=\\"background-color: #3b459a; display: inline-block; border: 2px solid transparent; color: #fafafa; padding: calc(.375em - 1px) .75em; text-align: center; white-space: nowrap; cursor: pointer; border-radius: 4px; box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4); font-size: 1rem; text-decoration: none; white-space: nowrap; height: 1.5em; vertical-align: top; line-height: 1.5; margin-right: 2rem !important;\\"><span>Reject Trade</span></a>
        </div>
      </div>
    </div>
    <footer style=\\"height: 5%; background: #4b528b; width: 75%; margin-left: auto; margin-right: auto;\\"></footer>
  </body>
</html>",
    "subject": "FlexFoxFantasy TradeMachine - Fantasy Baseball Trade Proposal from CREATOR_TEAM",
    "text": " TRADEMACHINE
CREATOR_TEAM REQUESTED A TRADE WITH YOU:
CREATOR_TEAM WOULD RECEIVE:

Squirtle Squad's 2020 1st round Low Minors pick (#12) from RECIPIENT_TEAM 
--------------------------------------------------------------------------------

RECIPIENT_TEAM WOULD RECEIVE:

Honus Wiener (Pirates - C, 1B) from CREATOR_TEAM 
MiniMe (Jays - P - Low Minors) from CREATOR_TEAM 
--------------------------------------------------------------------------------

Accept Trade [http://localhost:8080/trade/test-uuid/accept]Reject Trade
[http://localhost:8080/trade/test-uuid/reject]",
    "to": "test@example.com",
  },
}
`;

exports[`Emailer Class email snapshots sendTradeSubmissionEmail 1`] = `
Object {
  "envelope": Object {
    "from": "tradebot@flexfoxfantasy.com",
    "to": Array [
      "test@example.com",
    ],
  },
  "originalMessage": Object {
    "attachments": Array [],
    "from": "tradebot@flexfoxfantasy.com",
    "html": "<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\" style=\\"height: 100%; background: #f7f7f7;\\">
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width; initial-scale=1.0;\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta name=\\"x-apple-disable-message-reformatting\\">
    
    <style>
@media screen and (max-width: 600px) {
  u + .body {
    width: 100vw !important;
  }
}
</style>
    <style>
a.button-link {
  background-color: #3b459a;
  display: inline-block;
  border: 2px solid transparent;
  color: #fafafa;
  padding: calc(.375em - 1px) .75em;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  border-radius: 4px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4);
  font-size: 1rem;
  text-decoration: none;
  white-space: nowrap;
  height: 1.5em;
  vertical-align: top;
  line-height: 1.5;
}
@media only screen and (max-width:700px) {
  .main,
.header,
.footer {
    width: 100% !important;
  }

  .main .container h1 {
    font-size: 1.8rem !important;
  }

  a.button-link {
    margin: 1rem 0 !important;
    text-decoration: underline !important;
  }
}
</style>
    <style>
.button-container a.button-link {
  margin-right: 2rem !important;
}
</style>
    <title></title>
  </head>
  <body style=\\"font-family: -apple-system, BlinkMacSystemFont, Roboto, Helvetica, Arial, sans-serif; font-size: 16px; color: black; height: 100%; background: #f7f7f7;\\">
    <header class=\\"title\\" style=\\"height: 100px; background-color: #a9c07c; background-image: url('https://imgur.com/VKEupAm.jpg'); background-size: cover; background-position: center center; background-repeat: no-repeat; width: 75%; margin: auto;\\"></header>
    <div class=\\"main\\" style=\\"height: 80%; background: #fff; width: 75%; margin-left: auto; margin-right: auto;\\">
      <div class=\\"container\\" style=\\"width: 70%; margin-left: 10%; padding: 5% 0;\\">
        <h1 style=\\"line-height: initial; margin: 0.67em 0; padding: 0; color: #3b459a; font-weight: 600; font-size: 3rem; font-family: 'Lucida Grande', 'Montserrat', 'Roboto', monospace;\\">TradeMachine</h1>
        <h2>Your trade offer has been accepted by all recipients:</h2>
        <div>
          <h3>CREATOR_TEAM will receive:</h3>
          <ul class=\\"picks\\" style=\\"margin-top: 1em; margin-bottom: 2em;\\"></ul>
          <li>Squirtle Squad's 2020 1st round Low Minors pick (#12) from RECIPIENT_TEAM</li>
          <h3>RECIPIENT_TEAM will receive:</h3>
          <ul class=\\"majors\\" style=\\"margin-top: 1em; margin-bottom: 2em;\\"></ul>
          <li>Honus Wiener (Pirates - C, 1B) from CREATOR_TEAM</li>
          <ul class=\\"minors\\" style=\\"margin-top: 1em; margin-bottom: 2em;\\"></ul>
          <li>MiniMe (Jays - P - Low Minors) from CREATOR_TEAM</li>
        </div>
        <p style=\\"line-height: 1.6em; margin: 1em 0;\\">Upon submission of this trade, it will be announced in the <a href=\\"https://flexfoxfantasy.slack.com/archives/C84UJAX5Y\\" style=\\"text-decoration: none; border-bottom: #d3d6f0 1px solid; color: #2837b8 !important;\\">#trade_announcements</a> channel.</p>
        <div class=\\"button-container\\"><a class=\\"button-link\\" href=\\"http://localhost:8080/trade/test-uuid/submit\\" style=\\"background-color: #3b459a; display: inline-block; border: 2px solid transparent; color: #fafafa; padding: calc(.375em - 1px) .75em; text-align: center; white-space: nowrap; cursor: pointer; border-radius: 4px; box-shadow: 0 1px 3px rgba(0, 0, 0, 0.4); font-size: 1rem; text-decoration: none; white-space: nowrap; height: 1.5em; vertical-align: top; line-height: 1.5; margin-right: 2rem !important;\\"><span>Submit Trade</span></a></div>
      </div>
    </div>
    <footer style=\\"height: 5%; background: #4b528b; width: 75%; margin-left: auto; margin-right: auto;\\"></footer>
  </body>
</html>",
    "subject": "FlexFoxFantasy TradeMachine - Your Fantasy Baseball Trade Has Been Accepted!",
    "text": " TRADEMACHINE
YOUR TRADE OFFER HAS BEEN ACCEPTED BY ALL RECIPIENTS:
CREATOR_TEAM WILL RECEIVE:

Squirtle Squad's 2020 1st round Low Minors pick (#12) from RECIPIENT_TEAM 
RECIPIENT_TEAM WILL RECEIVE:

Honus Wiener (Pirates - C, 1B) from CREATOR_TEAM 
MiniMe (Jays - P - Low Minors) from CREATOR_TEAM Upon submission of this trade,
it will be announced in the #trade_announcements
[https://flexfoxfantasy.slack.com/archives/C84UJAX5Y] channel.

Submit Trade [http://localhost:8080/trade/test-uuid/submit]",
    "to": "test@example.com",
  },
}
`;
